src/openrct2/Game.cpp

```diff
@@ -565,7 +565,12 @@ void save_game()
 {
     if (!gFirstTimeSaving && !gIsAutosaveLoaded)
     {
-        const auto savePath = Path::WithExtension(gScenarioSavePath, ".park");
+        u8string savePath = gScenarioSavePath;
+        if (!String::Equals(Path::GetExtension(gScenarioSavePath), ".park", true))
+        {
+            savePath = Path::WithExtension(gScenarioSavePath, ".park");
```

Comment:
Doesn't this still replace the last dot with an extension? That said, I don't know how inputs look here exactly.

Reply:
The input has extension. This replace input like `.sv4` to `.park` but keep something like `.PARK`

Reply:
Unless they manage to load save without extension (which you cannot do from in game ui)